Mathlib/Order/Grade.lean

```diff
@@ -0,0 +1,371 @@
+/-
+Copyright (c) 2022 Yaël Dillies, Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Violeta Hernández Palacios, Grayson Burton, Vladimir Ivanov
+
+! This file was ported from Lean 3 source module order.grade
+! leanprover-community/mathlib commit 9003f28797c0664a49e4179487267c494477d853
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Data.Finset.Basic
+import Mathlib.Data.Int.SuccPred
+
+/-!
+# Graded orders
+
+This file defines graded orders, also known as ranked orders.
+
+A `𝕆`-graded order is an order `α` equipped with a distinguished "grade" function `α → 𝕆` which
+should be understood as giving the "height" of the elements. Usual graded orders are `ℕ`-graded,
+cograded orders are `ℕᵒᵈ`-graded, but we can also grade by `ℤ`, and polytopes are naturally
+`Fin n`-graded.
+
+Visually, `Grade ℕ a` is the height of `a` in the Hasse diagram of `α`.
+
+## Main declarations
+
+* `GradeOrder`: Graded order.
+* `GradeMinOrder`: Graded order where minimal elements have minimal grades.
+* `GradeMaxOrder`: Graded order where maximal elements have maximal grades.
+* `GradeBoundedOrder`: Graded order where minimal elements have minimal grades and maximal
+  elements have maximal grades.
+-- PORTING NOTE: not sure whether this should be called `grade` or `Grade`.
+* `Grade`: The grade of an element. Because an order can admit several gradings, the first argument
```

Comment:
```suggestion
* `grade`: The grade of an element. Because an order can admit several gradings, the first argument
```
This doesn't return a type

Reply:
do you mean "this should be in another section/not here" or "this should be lowercase"?

Reply:
This should indeed be lowercase.